src/components/editor-page/editor-pane/editor-pane.tsx

```diff
@@ -103,6 +98,29 @@ export const EditorPane: React.FC<EditorPaneProps & ScrollProps> = ({ onContentC
   const [editorScroll, setEditorScroll] = useState<ScrollInfo>()
   const onEditorScroll = useCallback((editor: Editor, data: ScrollInfo) => setEditorScroll(data), [])
 
+  const onPaste = useCallback((pasteEditor: Editor, event: PasteEvent) => {
+    if (!event || !event.clipboardData) {
+      return
+    }
+
+    // Check for pasted tables
+    const pasteText = event.clipboardData.getData('text')
+    if (smartPasteEnabled && pasteText && isTable(pasteText)) {
+      event.preventDefault()
+      extractTable(pasteText, pasteEditor)
+      return
+    }
+
+    // Upload pasted files
+    if (event.clipboardData.files && event.clipboardData.files.length > 0) {
+      event.preventDefault()
+      const files: FileList = event.clipboardData.files
+      if (files && files.length >= 1) {
+        handleUpload(files[0], pasteEditor)
+      }
+    }
```

Comment:
Instead of commenting I would extract these code blocks into separate functions

Reply:
Done